pydantic/_internal/_dataclasses.py

```diff
@@ -123,19 +124,25 @@ def complete_dataclass(
         typevars_map,
     )
 
-    # dataclass.__init__ must be defined here so its `__qualname__` can be changed since functions can't be copied.
+    # Get a temporary signature before we change the __init__
+    fields = cls.__pydantic_fields__  # type: ignore
+    temp_sig = generate_model_signature(cls.__init__, fields, config_wrapper)
```

Comment:
I think a cleaner approach here would be to modify the `generate_dataclass_signature` function to be more like the `generate_model_signature` function, instead of calling the `generate_model_signature` function in addition to `generate_dataclass_signature`.

please update

Reply:
I've made the changes.